### PR TITLE
Directly import `roboy_parser`

### PR DESCRIPTION
**Pull request by josephbirkner**
- Adjusted `SemanticSentenceAnalyser` to use new `SemanticAnalyserInterface` from roboy_parser.
- Moved runtime resources from parser to dialog.

#### TODO:
- [x] Update documentation
- [x] Install parser when installing dialog

### CODE REVIEW

**Comment by missxa** [comment]
@josephbirkner please rebase to devel and solve merge conflicts

File: src/main/java/roboy/linguistics/sentenceanalysis/SemanticParserAnalyzer.java [hunk 1]
```diff
@@ -59,100 +41,17 @@ public SemanticParserAnalyzer(int portNumber) {
...
-            } catch (IOException e) {
-                e.printStackTrace();
-                return interpretation;
-            }
-        } else
-            return interpretation;
+        String response;
+        SemanticAnalyzerInterface.Result result = semanticAnalyzer.analyze(
+            (String) interpretation.getFeature("sentence"));
+        interpretation.getFeatures().put(Linguistics.TOKENS, result.getTokens());
```
Comment (missxa) [inline src/main/java/roboy/linguistics/sentenceanalysis/SemanticParserAnalyzer.java]: so why are we not handling "(no answer)" responses?
why are you ignoring parser results like followupq?

**Review by missxa — CHANGES_REQUESTED** [review]
- i'd like to see a status check from our jenkins. please use the job WITHOUT MERGE

- doxygen strings are not updated

- i'm not so sure about the whole resources story. can we at leas make it clear which one belong to the parser? like a dedicated folder or so?

File: pom.xml [hunk 2]
```diff
@@ -372,6 +374,12 @@
             <version>4.2.1</version>
         </dependency>
 
+        <dependency>
```
Comment (nbasargin) [inline pom.xml]: So, parser is going to be a maven package?

File: src/main/java/roboy/linguistics/sentenceanalysis/SemanticParserAnalyzer.java [hunk 3]
```diff
@@ -59,100 +41,17 @@ public SemanticParserAnalyzer(int portNumber) {
...
+        interpretation.getFeatures().put(Linguistics.TRIPLE, extract_relations(result.getRelations()));
+        interpretation.getFeatures().put(Linguistics.SENTIMENT, result.getSentiment());
+        interpretation.getFeatures().put(Linguistics.POSTAGS, result.getPostags());
+        interpretation.getFeatures().put(Linguistics.LEMMAS, result.getLemmaTokens());
+        return interpretation;
```
Comment (nbasargin) [inline src/main/java/roboy/linguistics/sentenceanalysis/SemanticParserAnalyzer.java]: Do you plan to merge Wagram's Interpretation type-safety refactoring later?

File: start_submodules.sh [hunk 4]
```diff
@@ -1,28 +0,0 @@
-# start memory
```
Comment (nbasargin) [inline start_submodules.sh:1]: no start_submodules.sh anymore? cool

**Comment by Aevyz** [comment]
May want to consider adding https://github.com/Roboy/roboy_dialog/pull/39/commits/61e3b07011394276baabfaff7b47b18da7b9af43 to the branch --> Documentation

File: roboy_dialog/src/main/java/roboy/linguistics/sentenceanalysis/SemanticParserAnalyzer.java.orig [hunk 5]
```diff
@@ -59,6 +41,17 @@ public SemanticParserAnalyzer(int portNumber) {
      */
     @Override
     public Interpretation analyze(Interpretation interpretation) {
+<<<<<<< HEAD
```
Comment (missxa) [inline roboy_dialog/src/main/java/roboy/linguistics/sentenceanalysis/SemanticParserAnalyzer.java.orig]: merge conflict

**Review by missxa — CHANGES_REQUESTED** [review]

File: roboy_dialog/src/main/java/roboy/linguistics/sentenceanalysis/SemanticParserAnalyzer.java.orig (hunk 5 above)
Comment (josephbirkner) [reply roboy_dialog/src/main/java/roboy/linguistics/sentenceanalysis/SemanticParserAnalyzer.java.orig]: A .orig file that slipped into the commit... I'll remove it

**Pull request merged**